src/cargo/util/lints.rs

```diff
@@ -247,9 +247,39 @@ pub fn check_im_a_teapot(
     gctx: &GlobalContext,
 ) -> CargoResult<()> {
     let manifest = pkg.manifest();
+    let manifest_path = rel_cwd_manifest_path(path, gctx);
     let (lint_level, reason) = IM_A_TEAPOT.level(pkg_lints, ws_lints, manifest.edition());
 
     if IM_A_TEAPOT.is_feature_gated(manifest) {
+        if reason == LintLevelReason::Package || reason == LintLevelReason::Workspace {
```

Comment:
Add function to `reason` for this to make it easier to understand and evolve, like `is_user_specified`